website/server/controllers/api-v3/groups.js

```diff
@@ -402,6 +409,11 @@ api.leaveGroup = {
     }
 
     await group.leave(user, req.query.keep);
+
+    if (user.newMessages[group._id]) {
```

Comment:
Is there a reason to have this if check both here and in the removeMessages function?